.changeset/sour-coats-look.md

```diff
@@ -0,0 +1,5 @@
+---
+"@nomicfoundation/edr": patch
+---
+
+Adds EIP-7702 transactions to eth_call, eth_estimateGas, eth_sendTransaction, eth_sendRawTransaction, and debug_traceCall
```

Comment:
Question: support for `debug_traceCall` is not in the design doc. Is this intentional? If we support `debug_traceCall` should we support `debug_traceTransaction` as well? cc @fvictorio

Reply:
`debug_traceTransaction` receives the transaction hash of a previously mined transaction, so I didn't mention it explicitly. I tallied it up with methods like `eth_getTransactionByHash` which should also return an EIP-7702 transaction.

I can add some tests to validate that those succeed though.

Reply:
I added the tests I mentioned in https://github.com/NomicFoundation/edr/pull/805/commits/56a5820188862626d92a325e4c84fb1df869786c

Reply:
Maybe to expand on what Remco said: the authorizations part of EIP-7702 transactions is pretty much orthogonal to the execution part (similar to blob transactions, although blobs can be observed by the EVM; authorizations can't). So it makes sense for this to come for free without much extra effort.